### ISSUE TITLE
Circular reference when the `include` method is used in the `MATCH` command

### ISSUE THREAD

**Issue by dastoori** [orientechnologies/orientjs#271]
When I use the [`include`](https://orientdb.com/docs/2.2/SQL-Methods.html#include) method in the `MATCH` command, the output has circular reference, but when I write similar query with the `SELECT` command, the output is fine.

Examples:

**`MATCH` command:**

*(This is just an example, my real query is much more complex than this)*
```javascript
const stringify = require('json-stringify-safe')

const query = `
MATCH { class: User, as: user }
.out('UserProfile') { as: profile }
RETURN profile.include('name', 'age') as userInfo
`;

const result = await db.select().from(query).all();

console.log(stringify(result));
```

<details>
<summary><b>OrientJS output</b> (Click to open)</summary>
<pre>
[
  {
    "@type": "d",
    "userInfo": "[Circular ~.0]",
    "@rid": "#-1:-1",
    "@version": 0
  },
  {
    "@type": "d",
    "userInfo": {
      "@type": "d",
      "userInfo": "[Circular ~.1.userInfo]",
      "@rid": "#-1:-1",
      "@version": 0
    },
    "@rid": "#-1:-1",
    "@version": 0
  }
]
</pre>
</details>

<details>
<summary><b>Studio raw output</b> (Click to open)<br/><br/></summary>
<pre>
{
    "result": [
        {
            "@type": "d",
            "@version": 0,
            "userInfo": {
                "@type": "d",
                "@version": 0,
                "name": "John",
                "age": 32
            }
        },
        {
            "@type": "d",
            "@version": 0,
            "userInfo": {
                "@type": "d",
                "@version": 0,
                "name": "Janet",
                "age": 25
            }
        }
    ],
    "notification": "Query executed in 0.08 sec. Returned 2 record(s)"
}
</pre>
</details>

**`SELECT` command:**

```javascript
const query = `
SELECT out('UserProfile').include('name', 'age') as userInfo
FROM User
`;

const result = await db.select().from(query).all();

console.log(res);
```

<details>
<summary><b>OrientJS output</b> (Click to open)</summary>
<pre>
[
  {
    "@type": "d",
    "userInfo": [{
      "@type": "d",
      "name": "John",
      "age": 32
    }],
    "@rid": "#-2:0",
    "@version": 0
  },
  {
    "@type": "d",
    "userInfo": [{
      "@type": "d",
      "name": "Janet",
      "age": 15
    }],
    "@rid": "#-2:1",
    "@version": 0
  }
]
</pre>
</details>

<details>
<summary><b>Studio raw output</b> (Click to open)<br/><br/></summary>
<pre>
{
    "result": [
        {
            "@type": "d",
            "@rid": "#-2:0",
            "@version": 0,
            "userInfo": [
                {
                    "@type": "d",
                    "@version": 0,
                    "name": "John",
                    "age": 32
                }
            ],
            "@fieldTypes": "userInfo=z"
        },
        {
            "@type": "d",
            "@rid": "#-2:1",
            "@version": 0,
            "userInfo": [
                {
                    "@type": "d",
                    "@version": 0,
                    "name": "Janet",
                    "age": 25
                }
            ],
            "@fieldTypes": "userInfo=z"
        }
    ],
    "notification": "Query executed in 0.07 sec. Returned 2 record(s)"
}
</pre>
</details>

If you need more information, let me know.

Thanks

**Comment by dastoori** [comment]
@luigidellaquila, @maggiolo00, What is the status of this issue?

**Comment by maggiolo00** [comment]
hi @dastoori 

sorry for the late reply. I will try to reproduce it tomorrow and give you a feedback

Thanks

**Comment by maggiolo00** [comment]
Hi @dastoori 

the issue seems to be on the server. I'm working on a fix.

Thanks

**Comment by maggiolo00** [comment]
Hi @dastoori 

fixed. It will be available in the next GA release 2.2.25. It should be out soon.
